File: fe_coffee_manager/src/types/receiptStatus.ts
// Receipt Status Enums for Goods Receipt Management

export enum ReceiptStatus {
  // Basic statuses
  OK = 'OK',
  SHORT = 'SHORT',
  OVER = 'OVER',
  DAMAGE = 'DAMAGE',
  RETURN = 'RETURN',
  
  // Detailed statuses
  SHORT_ACCEPTED = 'SHORT_ACCEPTED',
  SHORT_PENDING = 'SHORT_PENDING',
  OVER_ACCEPTED = 'OVER_ACCEPTED',
  OVER_ADJUSTED = 'OVER_ADJUSTED',
  OVER_RETURN = 'OVER_RETURN',
  DAMAGE_ACCEPTED = 'DAMAGE_ACCEPTED',
  DAMAGE_RETURN = 'DAMAGE_RETURN',
  DAMAGE_PARTIAL = 'DAMAGE_PARTIAL'
}

export enum ReceiptStatusMessage {
  // Shortage messages
  ACCEPTED_SHORTAGE = '✅ ACCEPTED SHORTAGE',
  SHORT_PENDING = '⚠️ SHORT PENDING',
  
  // Overage messages
  ACCEPTED_OVERAGE = '✅ ACCEPTED OVERAGE',
  ADJUSTING_ORDER = '✅ ADJUSTING ORDER',
  
  // Damage messages
  FULL_DAMAGE_ACCEPTED = 'FULL DAMAGE ACCEPTED',
  RETURN_DAMAGED = 'RETURN DAMAGED',
  PARTIAL_DAMAGE_ACCEPTED = 'PARTIAL DAMAGE ACCEPTED'
}

export enum ReceiptStatusLabels {
  // Button labels
  ACCEPT_SHORTAGE = 'Accept Shortage',
  FOLLOW_UP_SHORTAGE = 'Follow Up',
  ACCEPT_OVERAGE = 'Accept Overage',
  ADJUST_ORDER = 'Adjust Order',
  RETURN_EXCESS = 'Return Excess',
  TAKE_GOOD_PARTS = 'Take Good Parts',
  RETURN_DAMAGED = 'Return Damaged',
  ACCEPT_FULL_DAMAGE = 'Accept Full Damage'
}

// Status mapping for UI display
export const StatusDisplayConfig = {
  [ReceiptStatus.SHORT_ACCEPTED]: {
    color: 'text-green-700',
    bgColor: 'bg-green-100',
    borderColor: 'border-green-300',
    icon: '✅'
  },
  [ReceiptStatus.SHORT_PENDING]: {
    color: 'text-orange-700',
    bgColor: 'bg-orange-100',
    borderColor: 'border-orange-300',
    icon: '⚠️'
  },
  [ReceiptStatus.OVER_ACCEPTED]: {
    color: 'text-green-700',
    bgColor: 'bg-green-100',
    borderColor: 'border-green-300',
    icon: '✅'
  },
  [ReceiptStatus.OVER_ADJUSTED]: {
    color: 'text-yellow-700',
    bgColor: 'bg-yellow-100',
    borderColor: 'border-yellow-300',
    icon: '📝'
  },
  [ReceiptStatus.OVER_RETURN]: {
    color: 'text-red-700',
    bgColor: 'bg-red-100',
    borderColor: 'border-red-300',
    icon: '↩️'
  },
  [ReceiptStatus.DAMAGE_ACCEPTED]: {
    color: 'text-orange-700',
    bgColor: 'bg-orange-100',
    borderColor: 'border-orange-300',
    icon: '⚡'
  },
  [ReceiptStatus.DAMAGE_RETURN]: {
    color: 'text-red-700',
    bgColor: 'bg-red-100',
    borderColor: 'border-red-300',
    icon: '↩️'
  },
  [ReceiptStatus.DAMAGE_PARTIAL]: {
    color: 'text-blue-700',
    bgColor: 'bg-blue-100',
    borderColor: 'border-blue-300',
    icon: '🔧'
  }
} as const;

File: catalog-service/src/main/java/com/service/catalog/enums/ReceiptStatus.java
package com.service.catalog.enums;

/**
 * Enum for Goods Receipt Status Management
 */
public enum ReceiptStatus {
    // Basic statuses
    OK("OK", "Normal receipt"),
    SHORT("SHORT", "Shortage detected"),
    OVER("OVER", "Overage detected"),
    DAMAGE("DAMAGE", "Damage detected"),
    RETURN("RETURN", "Items returned"),
    
    // Detailed statuses
    SHORT_ACCEPTED("SHORT_ACCEPTED", "Shortage accepted - no follow up needed"),
    SHORT_PENDING("SHORT_PENDING", "Shortage pending - follow up required"),
    OVER_ACCEPTED("OVER_ACCEPTED", "Overage accepted - keep all items"),
    OVER_ADJUSTED("OVER_ADJUSTED", "Order adjusted to match received quantity"),
    OVER_RETURN("OVER_RETURN", "Excess items returned to supplier"),
    DAMAGE_ACCEPTED("DAMAGE_ACCEPTED", "Damaged items accepted into inventory"),
    DAMAGE_RETURN("DAMAGE_RETURN", "Damaged items returned to supplier"),
    DAMAGE_PARTIAL("DAMAGE_PARTIAL", "Partial damage - only good items accepted");
    
    private final String code;
    private final String description;
    
    ReceiptStatus(String code, String description) {
        this.code = code;
        this.description = description;
    }
    
    public String getCode() {
        return code;
    }
    
    public String getDescription() {
        return description;
    }
    
    /**
     * Check if this status allows receiving more items
     */
    public boolean allowsMoreReceipts() {
        return this == SHORT_PENDING || this == OVER_ADJUSTED || this == DAMAGE_PARTIAL;
    }
    
    /**
     * Check if this status closes the receipt (no more receipts allowed)
     */
    public boolean closesReceipt() {
        return this == SHORT_ACCEPTED || this == OVER_ACCEPTED || this == DAMAGE_ACCEPTED;
    }
    
    /**
     * Get human-readable message for this status
     */
    public String getDisplayMessage() {
        switch (this) {
            case SHORT_ACCEPTED:
                return "✅ Shortage accepted - no follow up needed";
            case SHORT_PENDING:
                return "⚠️ Shortage pending - follow up required";
            case OVER_ACCEPTED:
                return "✅ Overage accepted - keeping all items";
            case OVER_ADJUSTED:
                return "📝 Order adjusted to match received quantity";
            case OVER_RETURN:
                return "↩️ Excess items returned to supplier";
            case DAMAGE_ACCEPTED:
                return "⚡ Damaged items accepted into inventory";
            case DAMAGE_RETURN:
                return "↩️ Damaged items returned to supplier";
            case DAMAGE_PARTIAL:
                return "🔧 Partial damage - only good items accepted";
            default:
                return description;
        }
    }
}
